compiler/noirc_frontend/src/parser/parser/traits.rs

```diff
@@ -28,11 +28,25 @@ pub(super) fn trait_definition() -> impl NoirParser<TopLevelStatement> {
         .then(ident())
         .then(function::generics())
         .then(where_clause())
-        .then_ignore(just(Token::LeftBrace))
-        .then(trait_body())
-        .then_ignore(just(Token::RightBrace))
+        .then(
+            just(Token::LeftBrace)
+                .ignore_then(trait_body())
+                .then_ignore(just(Token::RightBrace))
+                .or_not(),
+        )
         .validate(|((((attributes, name), generics), where_clause), items), span, emit| {
             let attributes = validate_secondary_attributes(attributes, span, emit);
+
+            let items = if let Some(items) = items {
+                items
+            } else {
+                emit(ParserError::with_reason(
+                    ParserErrorReason::ExpectedLeftBracketOrWhereOrLeftBraceOrArrowAfterTraitName,
+                    span,
+                ));
+                vec![]
+            };
```

Comment:
Seems like we could move this validation up to the `trait_body().or_not`? This would keep the validation closer to what we're validating. We could then break this out this parser into a `let trait_body_or_error =` (feel free to use a better name) above the main chain of parsers.

Reply:
Good idea, done!